tests/Translation/TranslationTranslatorTest.php

```diff
@@ -178,6 +178,14 @@ public function testGetJsonForNonExistingReturnsSameKeyAndReplaces()
         $this->assertEquals('foo baz', $t->getFromJson('foo :message', ['message' => 'baz']));
     }
 
+    public function testGetJsonForNonExistingJsonKeyDoesNotLookForRegularKeysIfKeyIsNotNested()
+    {
+        $t = new Translator($this->getLoader(), 'en');
+        $t->getLoader()->shouldReceive('load')->once()->with('en', '*', '*')->andReturn([]);
+        $t->getLoader()->shouldReceive('load')->with('en', 'foo', '*')->andReturn(['bar' => 'one']);
```

Comment:
tries to simulate the existence of a typical translation file with this content (lang/en/foo.php)